Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -893,10 +896,19 @@ - (void)fetchDefaultToken {
   [self defaultTokenWithHandler:handler];
 }
 
-- (void)defaultTokenWithHandler:(FIRInstanceIDTokenHandler)handler {
+- (void)defaultTokenWithHandler:(nullable FIRInstanceIDTokenHandler)aHandler {
+  // Store handlers to be called once operation completes
+  if (!self.defaultTokenFetchHandler) {
+    self.defaultTokenFetchHandler = [[FIRInstanceIDCombinedHandler<NSString *> alloc] init];
+  }
+  if (aHandler) {
+    [self.defaultTokenFetchHandler addHandler:aHandler];
+  }
+
   if (self.isFetchingDefaultToken || self.isDefaultTokenFetchScheduled) {
```

Comment:
Do we still need to check this flag or just check if defaultTokenFetchHandler.count > 1?

Reply:
Yeah, my first try was to replace `isFetchingDefaultToken` by checking `defaultTokenFetchHandler.count`, but then it is hard to handle retry logic due to the way it is implemented. Also, in this case the handler cannot be optional (which is the case e.g. in #2562). Keeping both, the flag and `defaultTokenFetchHandler` allows to make less changes to the existing logic.